src/types/message.ts

```diff
@@ -103,6 +79,33 @@ export interface StopMessage extends MessageBase {
   type: 'stop';
 }
 
+// Definitions for all messages from GCS to vehicles.
+export interface GeofenceMessage extends MessageBase {
+  type: 'geofence';
+
+  /**
+   * Name of job to perform.
+   */
+  jobType: JobType;
+
+  /**
+    * Geofencing coordinates in the form of a rectangle, and geofence type sent
+     along with start message.
+    */
+  geofence?: {
+    'keepOut': [number, number][],
+    'keepIn': [number, number][]
+  }
+}
+
+/**
+ * Type guard for Geofence Message.
+ */
+function isGeofenceMessage(message: Message): boolean {
+  return message.type === 'geofence'
+     && vehicleConfig.isValidJobType(message.jobType);
```

Comment:
this check is invalid

make the geofence field mandatory and check that the geofence field's keepIn and keepOut are arrays with 2 elements each (or just dont shouldnt matter since this message is from us)

adding this check will help me out in updating communication-js though